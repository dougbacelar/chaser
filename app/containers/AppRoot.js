import React from 'react';
import { StyleSheet, Text, View, AsyncStorage } from 'react-native';
import { connect } from 'react-redux';
import '../../global';
import AddressList from '../components/AddressList';

const Web3 = require('web3');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

async function test() {
  const myAddresses = {
    eth: ['1234567', 'abscasdqwe'],
    btc: ['654321'],
  };

  try {
    return AsyncStorage.setItem('addresses', JSON.stringify(myAddresses));
  } catch (err) {
    throw Error(`something went wrong yo ${err}`);
  }
}

export class AppRoot extends React.Component {
  componentWillMount() {
    test().then(AsyncStorage.getItem('addresses').then(console.log));

    const web3 = new Web3(
      new Web3.providers.HttpProvider('https://mainnet.infura.io/'),
    );

    // web3.eth.getBlock('latest').then(console.log);
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>Hello World!!</Text>
        <AddressList />
      </View>
    );
  }
}

export default connect(null, null)(AppRoot);
