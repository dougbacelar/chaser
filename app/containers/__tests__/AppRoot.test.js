import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { AppRoot } from '../AppRoot';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const component = shallow(<AppRoot />);

  expect(toJson(component)).toMatchSnapshot();
});
