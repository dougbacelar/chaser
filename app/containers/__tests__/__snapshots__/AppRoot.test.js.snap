// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Header
    backgroundColor="blue"
    centerComponent={
      Object {
        "style": Object {
          "color": "#fff",
        },
        "text": "Cryptofolio",
      }
    }
    rightComponent={
      Object {
        "color": "#fff",
        "icon": "add",
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    App Root!!
  </Text>
  <Connect(Home) />
</View>
`;
